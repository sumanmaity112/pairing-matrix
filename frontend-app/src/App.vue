<template>
  <div id="container">
    <PairingMatrixPage />
  </div>
</template>

<script>
import PairingMatrixPage from "@/components/PairingMatrixPage";

export default {
  name: "App",
  components: {
    PairingMatrixPage,
  },
};
</script>
