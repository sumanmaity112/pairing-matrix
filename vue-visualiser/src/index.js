import ChordPairingChart from "./components/ChordPairingChart";

export { ChordPairingChart };
