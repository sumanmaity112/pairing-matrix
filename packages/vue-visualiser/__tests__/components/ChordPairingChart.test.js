import { mount } from "@vue/test-utils";
import { ChordPairingChart } from "../../src/index";
import { pairingMatrix } from "../data/testData";

describe("ChordPairingChart", () => {
  afterEach(() => {
    document.body.innerHTML = "";
  });

  it("should render chord pairing chart", async () => {
    const wrapper = mount(ChordPairingChart, {
      props: {
        height: 700,
        width: 900,
        authors: pairingMatrix.authors,
        data: pairingMatrix.matrix,
      },
      attachTo: document.body,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should refresh chord pairing chart on data update", async () => {
    const initialData = pairingMatrix.matrix;
    const newData = {
      author: "joe",
      coAuthor: "tonystark",
      times: 1,
    };
    const updatedData = [...initialData, newData];
    const authors = pairingMatrix.authors;
    const wrapper = mount(ChordPairingChart, {
      props: { height: 700, width: 900, authors: authors, data: initialData },
      attachTo: document.body,
    });

    expect(wrapper.html()).toMatchSnapshot();

    await wrapper.setProps({
      data: updatedData,
      authors: [...authors, newData.author],
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
