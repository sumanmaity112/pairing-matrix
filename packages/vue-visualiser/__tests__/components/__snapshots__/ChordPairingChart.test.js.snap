// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChordPairingChart should refresh chord pairing chart on data update 1`] = `
<div id="chord-pairing-chart"><svg viewBox="-450,-350,900,700" svg-for="pairing-matrix-chord-chart">
    <g font-size="20" font-family="sans-serif">
      <g>
        <path fill="rgb(110, 64, 170)" d="M1.2858791391047208e-14,-210A210,210,0,0,1,15.735238527275941,209.40965180356355L14.985941454548515,199.43776362244148A200,200,0,0,0,1.2246467991473532e-14,-200Z"></path><text dy="0.35em" transform="
        rotate(-2.1485917317405807)
        translate(215)
        
      ">john.reese</text>
      </g>
      <g>
        <path fill="rgb(255, 140, 56)" d="M5.249453142089677,209.93437841789756A210,210,0,0,1,-176.39023770963516,113.958255693646L-167.9907025806049,108.53167208918666A200,200,0,0,0,4.99947918294255,199.9375032551405Z"></path><text dy="0.35em" transform="
        rotate(117.85140826825938)
        translate(215)
        rotate(180)
      " text-anchor="end">tonystark</text>
      </g>
      <g>
        <path fill="rgb(40, 234, 141)" d="M-181.8653347947321,105.00000000000007A210,210,0,0,1,-10.495625546842636,-209.73755468294289L-9.995833854135844,-199.75005207899324A200,200,0,0,0,-173.2050807568877,100.00000000000007Z"></path><text dy="0.35em" transform="
        rotate(208.56760551217292)
        translate(215)
        rotate(180)
      " text-anchor="end">zoe.morgan</text>
      </g>
    </g>
    <g fill-opacity="0.75">
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(40, 234, 141)" d="M0.4974994817709969,-198.9993781253239A199,199,0,0,1,177.32295580983302,90.31926340966254Q0,0,-163.91453957317748,94.09050811061248L-174.77244309675865,-95.16088027120317L-9.917942479691352,-188.73959419520202Q0,0,0.4974994817709969,-198.9993781253239Z" co-author="2" author="0">
        <title>john.reese → zoe.morgan 2</title>
      </path>
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(255, 140, 56)" d="M176.86914484210382,91.20474550604753A199,199,0,0,1,15.407066070626975,198.40267718731857Q0,0,4.252141204394723,188.9521613932423L-92.96884727715164,175.94826920421144L-158.49431203316536,102.95898723828638Q0,0,176.86914484210382,91.20474550604753Z" co-author="1" author="0">
        <title>john.reese → tonystark 1</title>
      </path>
    </g>
  </svg></div>
`;

exports[`ChordPairingChart should refresh chord pairing chart on data update 2`] = `
<div id="chord-pairing-chart"><svg viewBox="-450,-350,900,700" svg-for="pairing-matrix-chord-chart">
    <g font-size="20" font-family="sans-serif">
      <g>
        <path fill="rgb(110, 64, 170)" d="M1.2858791391047208e-14,-210A210,210,0,0,1,144.73409866918558,-152.15794649777072L137.8419987325577,-144.9123299978769A200,200,0,0,0,1.2246467991473532e-14,-200Z"></path><text dy="0.35em" transform="
        rotate(-68.21619724391353)
        translate(215)
        
      ">joe</text>
      </g>
      <g>
        <path fill="rgb(255, 94, 99)" d="M152.15794649777072,-144.73409866918556A210,210,0,0,1,10.495625546842424,209.7375546829429L9.995833854135643,199.75005207899326A200,200,0,0,0,144.9123299978769,-137.84199873255767Z"></path><text dy="0.35em" transform="
        rotate(21.783802756086473)
        translate(215)
        
      ">john.reese</text>
      </g>
      <g>
        <path fill="rgb(175, 240, 91)" d="M1.2858791391047208e-14,210A210,210,0,0,1,-209.7375546829429,10.495625546842438L-199.75005207899326,9.995833854135656A200,200,0,0,0,1.2246467991473532e-14,200Z"></path><text dy="0.35em" transform="
        rotate(133.56760551217295)
        translate(215)
        rotate(180)
      " text-anchor="end">tonystark</text>
      </g>
      <g>
        <path fill="rgb(26, 199, 194)" d="M-210,2.5717582782094416e-14A210,210,0,0,1,-10.49562554684245,-209.7375546829429L-9.995833854135668,-199.75005207899326A200,200,0,0,0,-200,2.4492935982947064e-14Z"></path><text dy="0.35em" transform="
        rotate(223.56760551217297)
        translate(215)
        rotate(180)
      " text-anchor="end">zoe.morgan</text>
      </g>
    </g>
    <g fill-opacity="0.75">
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(26, 199, 194)" d="M144.5301993760247,-136.79189109127248A199,199,0,0,1,144.5301993760247,136.79189109127248Q0,0,-188.9994093753076,-0.47249950781262057L-144.18776834788756,-137.15278873889483L-9.917942479691185,-188.73959419520202Q0,0,144.5301993760247,-136.79189109127248Z" co-author="3" author="1">
        <title>john.reese → zoe.morgan 2</title>
      </path>
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(175, 240, 91)" d="M0.4974994817709969,-198.9993781253239A199,199,0,0,1,136.79189109127248,-144.5301993760247Q0,0,-130.6026367608264,136.61607252120677L-180.86766532272077,82.99329876989025L-188.73959419520202,9.917942479691089Q0,0,0.4974994817709969,-198.9993781253239Z" co-author="2" author="0">
        <title>joe → tonystark 1</title>
      </path>
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(175, 240, 91)" d="M143.84443614666745,137.51282918203415A199,199,0,0,1,10.442701341050393,198.72581611029207Q0,0,-0.47249950781263217,188.9994093753076L-73.84996309670841,184.78956396564934L-129.91792671482662,137.26737528677722Q0,0,143.84443614666745,137.51282918203415Z" co-author="2" author="1">
        <title>john.reese → tonystark 1</title>
      </path>
    </g>
  </svg></div>
`;

exports[`ChordPairingChart should render chord pairing chart 1`] = `
<div id="chord-pairing-chart"><svg viewBox="-450,-350,900,700" svg-for="pairing-matrix-chord-chart">
    <g font-size="20" font-family="sans-serif">
      <g>
        <path fill="rgb(110, 64, 170)" d="M1.2858791391047208e-14,-210A210,210,0,0,1,15.735238527275941,209.40965180356355L14.985941454548515,199.43776362244148A200,200,0,0,0,1.2246467991473532e-14,-200Z"></path><text dy="0.35em" transform="
        rotate(-2.1485917317405807)
        translate(215)
        
      ">john.reese</text>
      </g>
      <g>
        <path fill="rgb(255, 140, 56)" d="M5.249453142089677,209.93437841789756A210,210,0,0,1,-176.39023770963516,113.958255693646L-167.9907025806049,108.53167208918666A200,200,0,0,0,4.99947918294255,199.9375032551405Z"></path><text dy="0.35em" transform="
        rotate(117.85140826825938)
        translate(215)
        rotate(180)
      " text-anchor="end">tonystark</text>
      </g>
      <g>
        <path fill="rgb(40, 234, 141)" d="M-181.8653347947321,105.00000000000007A210,210,0,0,1,-10.495625546842636,-209.73755468294289L-9.995833854135844,-199.75005207899324A200,200,0,0,0,-173.2050807568877,100.00000000000007Z"></path><text dy="0.35em" transform="
        rotate(208.56760551217292)
        translate(215)
        rotate(180)
      " text-anchor="end">zoe.morgan</text>
      </g>
    </g>
    <g fill-opacity="0.75">
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(40, 234, 141)" d="M0.4974994817709969,-198.9993781253239A199,199,0,0,1,177.32295580983302,90.31926340966254Q0,0,-163.91453957317748,94.09050811061248L-174.77244309675865,-95.16088027120317L-9.917942479691352,-188.73959419520202Q0,0,0.4974994817709969,-198.9993781253239Z" co-author="2" author="0">
        <title>john.reese → zoe.morgan 2</title>
      </path>
      <path style="stroke-width: 3px; mix-blend-mode: multiply;" fill="rgb(255, 140, 56)" d="M176.86914484210382,91.20474550604753A199,199,0,0,1,15.407066070626975,198.40267718731857Q0,0,4.252141204394723,188.9521613932423L-92.96884727715164,175.94826920421144L-158.49431203316536,102.95898723828638Q0,0,176.86914484210382,91.20474550604753Z" co-author="1" author="0">
        <title>john.reese → tonystark 1</title>
      </path>
    </g>
  </svg></div>
`;
